apps/arena/lib/arena/game_updater.ex

```diff
@@ -1848,6 +1867,16 @@ defmodule Arena.GameUpdater do
     game_state
   end
 
+  defp remove_pool_effects_from_entities(game_state, pool, entities) do
+    Enum.reduce(entities, game_state, fn {entity_id, _entity}, acc ->
+      if entity_id in pool.collides_with and pool.aditional_info.status == :READY do
```

Comment:
We're always checking if entity_id is in pool.collides_with. I think it's cheaper to just take the collided entities, apply the modifications, then merge (update) the entities map.

Reply:
Done with the refactor